modules/benchmark/R/calc.metrics.R

```diff
@@ -12,44 +11,74 @@
+  st <- format$vars$storage_type
   
-  # create results table
-  results <- as.data.frame(matrix(NA, nrow = length(vars_used$pecan_name), ncol = length(metrics$name)))
-  rownames(results) <- vars_used$pecan_name
-  colnames(results) <- metrics$name
+  time.row <- which(nchar(st)>1 & substr(st, 1,1) == "%")
   
-  # Do I need these?
-  # start.year <- lubridate::year(settings$run$start.date)
-  # end.year <- lubridate::year(settings$run$end.date)
+  # ---- MODEL DATA ---- #
   
-  model <- as.data.frame(read.output(runid=basename(model_run), outdir=model_run, start.year=start_year, end.year=end_year, format$vars$pecan_name))
-  # returns model output list object with variable values (including time), variable name and units
+  model_vars <- format$vars$pecan_name[-time.row]
+  model <- as.data.frame(read.output(runid=basename(model_run), outdir=model_run, 
+                                     start.year=start_year, end.year=end_year, 
+                                     c("time",model_vars)))
+  vars.used.index <- which(format$vars$pecan_name %in% names(model)[!names(model)=="time"])
   
-  vars_used <- as.data.frame(matrix(NA, nrow = length(names(model)), ncol = ncol(format$vars)))
-  colnames(vars_used) <- colnames(format$vars)
-  for(i in  1:length(names(model))){
-    vars_used[i,] <- format$vars[which(format$vars$pecan_name == names(model)[i]),]
+  # We know that the model output time is days since the beginning of the year.
+  # Make a column of years to supplement the column of days of the year.
+  years <- start_year:end_year
+  Diff <- diff(model$time)
+  n <- model$time[c(which(Diff < 0), length(model$time))]
+  y <- c()
+  for(i in 1:length(n)){
+    y <- c(y, rep(years[i], n[i]))
   }
+  model$year <- y
+  model$posix <- strptime(paste(model$time, model$year), format = "%j %Y")
+  model_full = model
   
-  obvs <- load.data(data.path, format, start_year, end_year, site=settings$run$site, vars_used)
-  # observation list object with variable values (including time), variable name and units
+  # ---- INPUT DATA ---- #
   
-  dat <- align.data(model, obvs, vars_used, start_year, end_year)
+  obvs <- load.data(data.path, format, start_year, end_year, site, vars.used.index, time.row)
+  dat_vars <- format$vars$pecan_name[vars.used.index]
+  obvs_full = obvs
   
-  for(v in 1:length(vars_used$pecan_name)){
-    metric_dat <- dat[,c(paste(vars_used$pecan_name[v], c("model", "obvs"),sep = "_" ),"time")]
+  #########################
+  
+  source("modules/benchmark/R/mean.over.larger.timestep.R")
+  dat <- align.data(model_full, obvs_full, dat_vars, start_year, end_year)
+  
+  results <- as.data.frame(matrix(NA, nrow = length(metrics$name), ncol = length(dat_vars)+1))
+  colnames(results) <- c("metric", dat_vars)
+  rownames(results) <- metrics$name
+  results$metric <- metrics$name
+  
+  file.sources = list.files("modules/benchmark/R/", pattern = "^metric.*", full.names = TRUE)
```

Comment:
If you want to search R functions, more generally, us 'apropos'. If you want to search this package specifically use ls("package:PEcAn.bechmark"). list.files will fail when this is a package